### ISSUE TITLE
Running a python custom plugin

### ISSUE THREAD

**Issue by frey0814** [NagiosEnterprises/ncpa#68]
I was trying to leverage the python engine built into the build by writing a custom plugin.  I wrote a very simple python script and named it test.py.

When I try to run the plugin via check_ncpa.py I get the following error:
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\flask\app.py", line 1504, in wsgi_app
  File "C:\Python27\lib\site-packages\flask\app.py", line 1264, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1262, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1248, in dispatch_request
  File "C:\ncpa\agent\listener\server.py", line 68, in decorated
  File "C:\ncpa\agent\listener\server.py", line 417, in api
  File "C:\ncpa\agent\listener\nodes.py", line 132, in run_check
  File "C:\ncpa\agent\listener\pluginnodes.py", line 51, in execute_plugin
  File "C:\Python27\lib\subprocess.py", line 709, in __init__
  File "C:\Python27\lib\subprocess.py", line 957, in _execute_child
WindowsError: [Error 193] %1 is not a valid Win32 application

Based on my research it looks like it is trying the run test.py directly.  I was hoping the ncpa detected the .py and would execute it as a python script.  To attempt to tell NCPA to launch it using python I added the following to the [plugin directives] section
.py = python $plugin_name $plugin_args

This unfortunately also didn't work and gave me the error:
WindowsError: [Error 2] The system cannot find the file specified

Which I guess is not too surprising.  

I would like to see python support to NCPA.  How hard would this be to add to NCPA?

**Comment by jhomann** [comment]
So after playing with NCPA for a while for all these fixes I think that it's safe to say cx_Freeze creates a python binary in the NCPA directory:

```
/usr/local/ncpa/Python
```

You can definitely run your own python scripts through this binary and it will use the frozen python that we include with NCPA. However, it doesn't have nearly the same amount of libraries as the standard python install on a system so it may not be that helpful. But you could definitely force scripts to use it.

**Comment by jhomann** [comment]
I'm actually going to close this since that is the way to do it by default.

**Comment by boneill81** [comment]
Was this issue ever addressed in the end? I'm on 2.1.5 and I can see from the changelog of 2.1.2 that an issue was addressed with python plug-ins not running correctly but it still does not seem to work for me. I wrote a small python dummy plug-in to test the functionality.

```PYTHON
import os, sys

OK = 0
WARNING  = 1
CRITICAL = 2
UNKNOWN  = 3

def main():

   print "OK: This is working"
   sys.exit(OK)

if __name__ == "__main__":
   main()
```
However this does not work. When run via check_ncpa.py on the command line from nagios core, it returns a 500 internal server error.

When executed via the api tester in the ncpa web server, it just stays at loading forever. Perhaps the functionality to execute custom python plugins via the bundled interpreter was never implemented, but I just wanted to check if this was the case since it would be amazingly useful if it were available.

Thanks.
